src/PromoService.kt

```
@@ -0,0 +1,1342 @@
+            // send Slack message for exhausted promos case
+            val errorMessage = "Custom Promo of type *${customPromoType.name}* not available for user with Id: *$userId*"
+            logger.critical(errorMessage, AlertType.SLACK)
+            throw CustomPromoNotFoundException()
+        }
+        return newFlashCoffeePromos
+    }
+
+    fun saveCustomPromos(userId: Long, customPromos: List<CustomPromo>) {
+        try {
+            customPromoRepository.saveAll(customPromos)
+        } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)